cargo-marker/src/cli.rs

```diff
@@ -1,115 +1,161 @@
+        .collect();
 
-        let Config { lints } = Config::try_from_str(&virtual_manifest).map_err(ConfigFetchError::emit_and_convert)?;
-        Ok(lints)
+    let cargo_args = if let Some(index) = marker_args.iter().position(|val| val == CARGO_ARGS_SEPARATOR) {
+        let mut cargo_args = marker_args.split_off(index);
+        // Remove `CARGO_ARGS_SEPARATOR` from the start
+        cargo_args.remove(0);
+        cargo_args
     } else {
-        Err(ExitStatus::NoLints)
-    }
+        vec![]
+    };
+
+    (marker_args, cargo_args)
 }
 
-pub fn get_clap_config() -> Command {
-    Command::new(VERSION)
-        .arg(
-            Arg::new("version")
-                .short('V')
-                .long("version")
-                .action(ArgAction::SetTrue)
-                .help("Print version info and exit"),
-        )
-        .arg(
-            Arg::new("verbose")
-                .short('v')
-                .long("verbose")
-                .action(ArgAction::SetTrue)
-                .help("Print additional debug information to the console"),
-        )
-        .arg(
-            Arg::new("test-setup")
-                .long("test-setup")
-                .action(ArgAction::SetTrue)
-                .help("This flag will compile the lint crate and print all relevant environment values"),
-        )
-        .arg(
-            Arg::new("forward-rust-flags")
-                .long("forward-rust-flags")
-                .action(ArgAction::SetTrue)
-                .help("Forwards the current `RUSTFLAGS` value during driver and lint compilation"),
-        )
-        .subcommand(setup_command())
-        .subcommand(check_command())
-        .args(check_command_args())
-        .after_help(AFTER_HELP_MSG)
-        .override_usage("cargo-marker [OPTIONS] -- <CARGO ARGS>")
+#[derive(Subcommand, Debug)]
+pub enum CliCommand {
+    /// Run Marker on the current package
+    Check(CheckArgs),
+    /// Setup the rustc driver for Marker
+    Setup(SetupArgs),
+    /// **UNSTABLE** Setup the specified lint crate for ui tests
+    #[command(hide = true)]
+    TestSetup(CheckArgs),
 }
 
-fn setup_command() -> Command {
-    Command::new("setup")
-        .about("A collection of commands to setup marker")
-        .after_help("By default this will install the driver for rustc.")
-        .arg(
-            Arg::new("auto-install-toolchain")
-                .long("auto-install-toolchain")
-                .action(ArgAction::SetTrue)
-                .help("This automatically installs the required toolchain using rustup"),
-        )
+#[derive(Args, Debug)]
+#[command(override_usage = "cargo marker check [OPTIONS] -- <CARGO ARGS>")]
+pub struct CheckArgs {
+    /// Specifies lint crates which should be used. (Lints in `Cargo.toml` will be ignored)
+    #[arg(short, long)]
+    pub lints: Vec<String>,
+    /// Forwards the current `RUSTFLAGS` value during driver and lint crate compilation
+    #[arg(long, default_value_t = false)]
```

Comment:
IIRC booleans have the `false` default by default. I think you don't need to explicitly spell `default_value_t` or maybe something changed in the new version of clap? Because I am still stuck with `structopt` on my work 😓

Reply:
I'll try this. The PR was hacked together during travels with a bad internet connection ^^